angular-ngrx-scss/src/app/shared/components/repo-controls/repo-controls.component.ts

```diff
@@ -1,11 +1,61 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { FormControl } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import {
+  debounceTime,
+  distinctUntilChanged,
+  combineLatest,
+  Subject,
+} from 'rxjs';
+import { startWith, takeUntil } from 'rxjs/operators';
+import { setSortAndFilterProperties } from 'src/app/state/profile/profile.actions';
+import {
+  hasActiveSortAndFilters,
+  selectFilterBySearch,
+  selectReposCount,
+} from 'src/app/state/profile/profile.selectors';
+import { SortAndFilterState } from 'src/app/state/profile/profile.state';
 
 @Component({
   selector: 'app-repo-controls',
   templateUrl: './repo-controls.component.html',
   styleUrls: ['./repo-controls.component.scss'],
 })
-export class RepoControlsComponent {
+export class RepoControlsComponent implements OnInit, OnDestroy {
+  private destroy$ = new Subject();
```

Comment:
Curious, is this easier if we make this a simply boolean prop?

Reply:
it was a recommendation from @Reshurum in other comment => https://github.com/thisdot/starter.dev-github-showcases/pull/463#discussion_r920403383

We are cleaning the Observables using that way.